app/services/ecr_service.rb

```diff
@@ -0,0 +1,53 @@
+class EcrService
+  def initialize(heritage)
+    @heritage = heritage
+  end
+
+  def validate_image!
+    begin
+      response = ecr.describe_images({
+                                       image_ids: [
+                                         {
+                                           image_tag: @heritage.tag
+                                         }
+                                       ],
+                                       repository_name: repository_name
+                                     })
+    rescue Aws::ECR::Errors::RepositoryNotFoundException,
+           Aws::ECRPublic::Errors::RepositoryNotFoundException
+      raise ExceptionHandler::BadRequest.new("Image not found in ECR: #{@heritage.image_path}")
+    rescue Aws::ECR::Errors::ImageNotFoundException,
+           Aws::ECRPublic::Errors::ImageNotFoundException
+      raise ExceptionHandler::BadRequest.new("Image tag not found in ECR: #{@heritage.image_path}")
+    end
+  end
+
+  private
+
+  def ecr
+    @ecr ||= if public_ecr?
+               ecr_public
+             else
+               ecr_private
+             end
+  end
+
+  def ecr_public
+    @heritage.district.aws.public_ecr
+  end
+
+  def ecr_private
+    @heritage.district.aws.ecr
+  end
```

Comment:
I htink you can move this into the district class, instead of using the `aws` property directly inside

```suggestion
  @ecr ||= heritage.district.ecr
```

Reply:
updated it https://github.com/degica/barcelona/pull/703/commits/67214810305cc5efe68a46d39e4de45ef60098e5